core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/util/QueryEvaluationUtil.java

```diff
@@ -123,9 +129,48 @@ private static boolean valuesEqual(Value leftVal, Value rightVal) {
 		return leftVal != null && rightVal != null && leftVal.equals(rightVal);
 	}
 
+	/**
+	 * Compares the supplied {@link Literal} arguments using the supplied operator, using strict
+	 * (minimally-conforming) SPARQL 1.1 operator behavior.
+	 * 
+	 * @param leftLit
+	 *        the left literal argument of the comparison.
+	 * @param rightLit
+	 *        the right literal argument of the comparison.
+	 * @param operator
+	 *        the comparison operator to use.
+	 * @return {@code true} if execution of the supplied operator on the supplied arguments succeeds,
+	 *         {@code false} otherwise.
+	 * @throws ValueExprEvaluationException
+	 *         if a type error occurred.
+	 */
 	public static boolean compareLiterals(Literal leftLit, Literal rightLit, CompareOp operator)
 		throws ValueExprEvaluationException
 	{
+		return compareLiterals(leftLit, rightLit, operator, true);
+	}
+
+	/**
+	 * Compares the supplied {@link Literal} arguments using the supplied operator.
+	 * 
+	 * @param leftLit
+	 *        the left literal argument of the comparison.
+	 * @param rightLit
+	 *        the right literal argument of the comparison.
+	 * @param operator
+	 *        the comparison operator to use.
+	 * @param strict
+	 *        boolean indicating whether comparison should use strict (minimally-conforming) SPARQL 1.1
+	 *        operator behavior, or extended behavior.
+	 * @return {@code true} if execution of the supplied operator on the supplied arguments succeeds,
+	 *         {@code false} otherwise.
+	 * @throws ValueExprEvaluationException
+	 *         if a type error occurred.
+	 */
+	public static boolean compareLiterals(Literal leftLit, Literal rightLit, CompareOp operator,
+			boolean strict)
```

Comment:
If we start making finer distinctions we can replace the boolean with something like an enum, I guess. But I prefer not to over-engineer at this point - it seems unlikely to me that we will very quickly implement fine-grained picking-and-choosing of extended behaviors. For one thing it will make testing complex, and for another, it puts a burden on third-party Sail implementors. So all for two flavors for now: either strict, or extended, where extended is just whatever we feel is useful.